### DISCUSSION TITLE
Redesign

### DISCUSSION THREAD

**Discussion by jarvis394** [jarvis394/geekr#78] [General]
Сюда буду скидывать прогресс редизайна.

<div>
<img width="256px" src="https://user-images.githubusercontent.com/37776763/103808977-ddecbd80-5069-11eb-96be-dbeaf2100ce8.jpg" />
<img width="256px" src="https://user-images.githubusercontent.com/37776763/103808982-df1dea80-5069-11eb-9bd6-481bf7517512.jpg" />
</div>

**Comment by jarvis394** [comment]
Скринкаст нового приложения:

https://photos.app.goo.gl/b3ahZWnQ5YpqECxD6

**Comment by jarvis394** [comment]
~~Теперь редизайн хостится онлайн здесь.~~
[habra.vercel.app](https://habra.vercel.app) 

Можно посмотреть, потыкать, поизучать. Предупрждаю, **это ещё далеко не финальный вариант!** Если у вас лагает сайт, это из-за того, что я не оптимизировал компоненты и также из-за того, что сайт запущен в development режиме.

**Comment by jarvis394** [comment]
Сайт переехал на новый хост: [habra.vercel.app](https://habra.vercel.app)

**Comment by jarvis394** [comment]
<div>
  <img width="256px" src="https://user-images.githubusercontent.com/37776763/117551357-38188200-b04e-11eb-8695-f04ca8857f47.png" />
  <img width="256px" src="https://user-images.githubusercontent.com/37776763/117551356-36e75500-b04e-11eb-8590-fc165370f9cb.png" />
</div>

Не очень мне нравится текущая домашняя страница, т.к. её очень сложно оптимизировать под десктоп :(
Вот и переделал.

панель снизу старая,  если что :p

**Comment by LencoDigitexer** [comment]
![image](https://user-images.githubusercontent.com/44779327/124361169-6376b200-dc36-11eb-8b11-aa87aec0a4a3.png)
Может, сделать строку заголовка белой в белой, а темную в темной? 

`<meta name="theme-color" media="(prefers-color-scheme: light)" content="white">`
`<meta name="theme-color" media="(prefers-color-scheme: dark)" content="black">`

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta/name/theme-color

**Reply by jarvis394** [reply]
> Может, сделать строку заголовка белой в белой, а темную в темной?

Не совсем понял, что к чему 😟
Может, картинкой объясните?

**Reply by jarvis394** [reply]
Оформление окна вроде как нельзя менять (я по крайней мере в спецификации не видел). Windows скма похоже находит ключевые цвета и красит окно.

**Comment by LencoDigitexer** [comment]
Можно попробовать на jquery сделать(знаю, что не используешь его, но на StackOverFLow вроде было решение https://stackoverflow.com/questions/48330535/dynamically-add-meta-description-based-on-route-in-angular)

```
$("meta[name='theme-color']").attr("content", "#000");
$("meta[name='msapplication-navbutton-color']").attr("content", "#000");
$("meta[name='viewport']").attr("content", "ya-title=#000,ya-dock=fade");
```

Либо на чистом JS -
https://developer.mozilla.org/ru/docs/Web/API/Element/setAttribute


P.S. Только что внедрил эту фичу на свой сайт, поэтому вспомнил про хабру

**Comment by LencoDigitexer** [comment]
Вариант, если используются cookie
```javascript
    <!--theme chrome for browser-->
    <meta name="theme-color" content="#FFFFFF" />
    <meta name="msapplication-navbutton-color" content="#FFFFFF" />
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-transcluent" />
    <meta name="viewport" content="ya-title=#FFFFFF,ya-dock=fade" />
    <script>
        function getCookie(name) { // функция, которая берет значение куки по имени
            let matches = document.cookie.match(new RegExp(
                "(?:^|; )" + name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, '\\$1') + "=([^;]*)"
            ));
            return matches ? decodeURIComponent(matches[1]) : undefined;
        }
        var theme = getCookie("dark") //берём куки вида dark=on (темная тема), или dark=off (светлая тема)
        if(theme == "on"){ // если тёмная тема, то меняем значение
            document.getElementsByName("theme-color")[0].setAttribute("content", "#101924");
            document.getElementsByName("msapplication-navbutton-color")[0].setAttribute("content", "#101924");
            document.getElementsByName("viewport")[0].setAttribute("content", "ya-title=#101924,ya-dock=fade");
        } // иначе значение будет неизменным, которое прописано в meta
 
```

**Comment by jarvis394** [comment]
https://www.figma.com/file/YxGtGOEkmWyJA32OjhG5KG/geekr

![image](https://user-images.githubusercontent.com/37776763/135724471-a0d55904-8734-4e8b-936c-81c8e73680f2.png)

**Reply by LencoDigitexer** [reply]
Отличный дизайн. В тренде, прям как Яндекс, например

**Reply by jarvis394** [reply]
пока что надо подумать, как это оптимизировать под десктоп 🤔
